lib/streamlit/runtime/caching/hashing.py

```diff
@@ -267,7 +267,7 @@ def _to_bytes(self, obj: Any) -> bytes:
             return self.to_bytes(dataclasses.asdict(obj))
 
         elif isinstance(obj, Enum):
-            return (str(obj._value_) + str(obj._name_)).encode()
+            return str(obj).encode()
```

Comment:
This should be enough for a unique hash: 

```
        class EnumClass(Enum):
            ENUM_1 = "hello"
            ENUM_2 = "world"

        str(EnumClass.ENUM_1) // should return "EnumClass.ENUM_1" string
```

Previous implementation potentially could return similar values for different Enum class members with same name and values. Old hashing implementation surprisingly passes new test, but my test snippet shoues that could be false: https://replit.com/join/mcfyoealue-kajarenc